bigbluebutton/greenlight
# Create room feature

TODO:
    - Add the capability to define the room name.



## Description

This PR enables creating a room for an authenticated user through GL-v3 app.
The authenticated user access the rooms area.
In the rooms area the user click the `+ new room` button and a new room will get created with a name of `Room` and a prefix of the timestamp of the creation event on client side.
**Note**: Made some minor changes to the RoomCard component styling to fix the card size in an expected way.
## Testing Steps


> 1. Pull the code.
> 2. Install the dependencies `bundle install && npm|yarn install`.
> 3. Clean the previous assets build by running `rm app/assets/builds/*` (This won't remove .keep since it's hidden).
> 4. Clean the database and tmp files for a better isolation by running `rails tmp:clear && rails db:schema:cache:clear && rails db:migrate:reset`
> 5. Run the linter and specs `bundle exec rubocop --parallel && bundle exec rspec`
> 6. Run `./bin/dev` to run the assets builders processes and the Puma server all at once.

1. Login with a pre created account.
2. Access the `/rooms` page.
3. Click on the `+ New Room` button
4. Confirm that:
  4.1  A new room card gets created as the last one in the rooms list.
  4.2 The new room card will be a placeholder card having:
     4.2.1 The same size as the other room cards.
     4.2.2 The same consistent format (the button , the horizontal rule, the icons, the text placeholders actual placement ).
     4.2.3 No effect once clicked or hovered over.
     4.2.4 Having no effect if its `start` button gets clicked.
  4.3 The `+ New Room` button gets disabled and a spinner with growing animation gets displayed inside it.
  4.4 Once the `+ New Room` animation stops, the spinner disappears and the button becomes enabled (clickable):
     4.4.1 The new room placeholder becomes a valid room card with the room's data matching those in the db.
     4.4.2 The new room card will be clickable and have highlighting effects once hovered over.
     4.4.3 Clicking on the new room card will have the expected outcome as any other room card. 
 -> Any error that pops during the process should revert the rooms list that the page had before committing the action of creating the new room along with a full re-fetch to the cache (rooms list) after reverting.
-> To better visualize the animations kindly consider throttling the network speed in the Dev tools.

## Screenshots (if appropriate):


![image](https://user-images.githubusercontent.com/29759616/163412663-d92a18fa-35e4-4747-ae8e-201c1da6711d.png)

![image](https://user-images.githubusercontent.com/29759616/163412757-4d4bde3a-6c8f-4ad5-b35c-74ea1ea194ff.png)

![image](https://user-images.githubusercontent.com/29759616/163412932-626e10ca-da44-49b9-9bb3-b5795b6fb161.png)

![image](https://user-images.githubusercontent.com/29759616/163413035-d27891e6-a9b4-4c5d-9a0d-f082da9d28aa.png)

![image](https://user-images.githubusercontent.com/29759616/163413108-f44c289f-0efe-47d5-b807-d8063889e6cd.png)
